providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/connector/file_watcher.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+import os
+import threading
+import time
+import typing
+
+import yaml
+
+from openfeature.evaluation_context import EvaluationContext
+from openfeature.event import ProviderEventDetails
+from openfeature.exception import ParseError, ProviderNotReadyError
+from openfeature.provider.provider import AbstractProvider
+
+from ..connector import FlagStateConnector
+from ..flags import FlagStore
+
+logger = logging.getLogger("openfeature.contrib")
+
+
+class FileWatcher(FlagStateConnector):
+    def __init__(
+        self,
+        file_path: str,
+        provider: AbstractProvider,
+        flag_store: FlagStore,
+        poll_interval_seconds: float = 1.0,
+    ):
+        self.file_path = file_path
+        self.provider = provider
+        self.poll_interval_seconds = poll_interval_seconds
+
+        self.last_modified = 0.0
+        self.flag_store = flag_store
+        self.should_emit_ready_on_success = False
+
+    def initialize(self, evaluation_context: EvaluationContext) -> None:
+        self.active = True
+        self.thread = threading.Thread(target=self.refresh_file, daemon=True)
```

Comment:
Why is daemon set to true in this thread?

Reply:
Let's add a name to all child threads for easier debugging. This could be `FlagdFileWatcherWorkerThread`

Reply:
Basically this:
> By setting them as daemon threads, you can let them run and forget about them, and when your program quits, any daemon threads are killed automatically.

https://stackoverflow.com/questions/190010/daemon-threads-explanation